mlir-sycl/include/mlir/Dialect/SYCL/Analysis/AliasAnalysis.h

```diff
@@ -21,6 +21,10 @@ namespace sycl {
 
 /// Specialized alias analysis for SYCL dialect operations.
 class AliasAnalysis : public LocalAliasAnalysis {
+public:
+  AliasAnalysis(bool relaxedAliasing)
+      : LocalAliasAnalysis(), relaxedAliasing(relaxedAliasing) {}
```

Comment:
I'm a bit worried this could cause issues. I don't see other analyses getting options and, looking at how the analysis manager works ([see](https://mlir.llvm.org/docs/PassManagement/#analysis-management)), caching an analysis with a given option might cause issues (querying `AliasAnalysis`, how do we know whether the option has been passed?). I suggest we make this option transparent to the analysis manager by creating a three-class hierarchy here: 
- `AliasAnalysis`: Base, will hold the option, hidden
- `AliasAnalysisStrict`
- `AliasAnalysisRelaxed`

This way, we know that we're querying either `AliasAnalysisStrict` or `AliasAnalysisRelaxed`.

Reply:
I don't think it would cause issues because passes that use the SYCL AliasAnalysis it would always use the same one. That is, for a given compilation unit, there can only be one setting for the -relaxed-aliasing option.

Having to create derived classes like proposed above is not scalable, if we had 2 options then we would need 4 classes and so on.

Reply:
Yes, in fact, I think it does not hurt. I was misunderstanding how the cache works. LGTM.